app/helpers/application_helper.rb

```diff
@@ -16,13 +16,7 @@ def step_class(step, active)
   end
 
   def tooltip(text)
-    content_tag(
-      :span, \
-      image_tag(asset_url('tooltip.svg'), width: 16, class: 'px1 img-tooltip'), \
-      class: 'hint--top hint--no-animate', \
-      'aria-label': text, \
-      'tabindex': '0',
-    )
+    render('shared/help-tooltip') { text }
```

Comment:
I think the block is kind of overkill here? Since the `tooltip` helper method doesn't take a block, there's no reason to use it? We could just pass the text as a local?

```suggestion
    render 'shared/help-tooltip', locals: { text: text }
```

Reply:
> I think the block is kind of overkill here? Since the `tooltip` helper method doesn't take a block, there's no reason to use it? We could just pass the text as a local?

It's a good point, though my initial reaction isn't to remove the block rendering, but to bypass / remove the `tooltip` helper and render the partial directly where it's used. Curious if you have any recommendations or resources on when and when not to use blocks. In my mind, I'd been treating it as analogous to attributes vs. inner content of an HTML element, or props vs. children of a React component.

FWIW, I'd originally implemented it using a local, and in later revisions opted to refactor to a block. Fine to switch back though.

Reply:
I would also be a fan of removing the helper and using the partial directly

Re: when to use blocks vs not.... I'll start poking around but the main cases I want to use blocks are:
- deferred evaluation (expensive operation we may not want to do immediately)
- repeated evaluation (such as a loop)
- syntax advantage
- adding in custom methods or scope (ex a mini-DSL)
- lifecycle/resource management (ex custom cleanup the method can call after before/after the block)

I think the biggest case we could make here is with the syntax advantage of having the block structure here which matches nesting HTML tags

Reply:
Thanks for sharing that list of examples. Those make a lot of sense, and it's clearer to me now why you'd flagged it in the first place, since I can see that there's not a whole lot of benefit to implementing it as a block if we're only ever using it through this view helper.

I'll plan to eliminate the helper. Given your comment, I'll assume it'd be reasonable to keep it as accepting a block once we're rendering the partial directly.

Reply:
> I'll plan to eliminate the helper. Given your comment, I'll assume it'd be reasonable to keep it as accepting a block once we're rendering the partial directly.

Done in 38041df.